sqlmesh/core/context.py

```diff
@@ -624,21 +624,27 @@ def get_model(
             The expected model.
         """
         if isinstance(model_or_snapshot, str):
-            normalized_name = normalize_model_name(
-                model_or_snapshot,
-                dialect=self.default_dialect,
-                default_catalog=self.default_catalog,
-            )
-            model = self._models.get(normalized_name)
+            # We should try all dialects referenced in the project for cases when models use mixed dialects.
+            all_dialects = {m.dialect for m in self._models.values() if m.dialect} | {
```

Comment:
cache this otherwise get_model becomes n^2

Reply:
I don't think we ever use `get_model` in a loop, do we?

Reply:
no but it could be